src/model/test/CoilCoolingDXSingleSpeedThermalStorage_GTest.cpp

```diff
@@ -0,0 +1,859 @@
+  // ASSERT_TRUE(coilCoolingDXSingleSpeedThermalStorage.storageTankPlantConnectionDesignFlowRate());
+  // EXPECT_EQ(10.4, coilCoolingDXSingleSpeedThermalStorage.storageTankPlantConnectionDesignFlowRate().get());
+  // Bad Value
+  // EXPECT_FALSE(coilCoolingDXSingleSpeedThermalStorage.setStorageTankPlantConnectionDesignFlowRate(-10.0));
+  // ASSERT_TRUE(coilCoolingDXSingleSpeedThermalStorage.storageTankPlantConnectionDesignFlowRate());
+  // EXPECT_EQ(10.4, coilCoolingDXSingleSpeedThermalStorage.storageTankPlantConnectionDesignFlowRate().get());
+
+  // Storage Tank Plant Connection Heat Transfer Effectiveness: Required Double
+  // EXPECT_TRUE(coilCoolingDXSingleSpeedThermalStorage.setStorageTankPlantConnectionHeatTransferEffectiveness(0.991));
+  // EXPECT_EQ(0.991, coilCoolingDXSingleSpeedThermalStorage.storageTankPlantConnectionHeatTransferEffectiveness());
+  // Bad Value
+  // EXPECT_FALSE(coilCoolingDXSingleSpeedThermalStorage.setStorageTankPlantConnectionHeatTransferEffectiveness(-10.0));
+  // EXPECT_EQ(0.991, coilCoolingDXSingleSpeedThermalStorage.storageTankPlantConnectionHeatTransferEffectiveness());
+
+  // Storage Tank Minimum Operating Limit Fluid Temperature: Optional Double
+  // EXPECT_TRUE(coilCoolingDXSingleSpeedThermalStorage.setStorageTankMinimumOperatingLimitFluidTemperature(10.6));
+  // ASSERT_TRUE(coilCoolingDXSingleSpeedThermalStorage.storageTankMinimumOperatingLimitFluidTemperature());
+  // EXPECT_EQ(10.6, coilCoolingDXSingleSpeedThermalStorage.storageTankMinimumOperatingLimitFluidTemperature().get());
+
+  // Storage Tank Maximum Operating Limit Fluid Temperature: Optional Double
+  // EXPECT_TRUE(coilCoolingDXSingleSpeedThermalStorage.setStorageTankMaximumOperatingLimitFluidTemperature(10.7));
+  // ASSERT_TRUE(coilCoolingDXSingleSpeedThermalStorage.storageTankMaximumOperatingLimitFluidTemperature());
+  // EXPECT_EQ(10.7, coilCoolingDXSingleSpeedThermalStorage.storageTankMaximumOperatingLimitFluidTemperature().get());
+}
+TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_HeatCoolFuelTypes) {
+  Model m;
+
+  CoilCoolingDXSingleSpeedThermalStorage coilCoolingDXSingleSpeedThermalStorage(m);
+
+  EXPECT_EQ(ComponentType(ComponentType::Cooling), coilCoolingDXSingleSpeedThermalStorage.componentType());
+  testFuelTypeEquality({FuelType::Electricity}, coilCoolingDXSingleSpeedThermalStorage.coolingFuelTypes());
+  testFuelTypeEquality({}, coilCoolingDXSingleSpeedThermalStorage.heatingFuelTypes());
+  testAppGFuelTypeEquality({}, coilCoolingDXSingleSpeedThermalStorage.appGHeatingFuelTypes());
+}
+
+TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_containingHVACComponent) {}
+
+TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_containingZoneHVACComponent) {}
+
+TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_addToNode) {}
+
+TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_clone) {}
+
+TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_cloneParent) {}
```

Comment:
I came looking for these. Need to implement them.

Reply:
Implemented.